test/test_factorizations.jl

```diff
@@ -0,0 +1,62 @@
+using Test: @test, @testset, @inferred
+using TestExtras: @constinferred
+using TensorAlgebra: contract, svd, tsvd, TensorAlgebra
+using TensorAlgebra.MatrixAlgebraKit: truncrank
```

Comment:
Just as a style preference, I prefer `using MatrixAlgebraKit: truncrank`. `using TensorAlgebra.MatrixAlgebraKit` implicitly assumes the dependency is part of the API of the package, which I think isn't a good habit to get into and also just gets confusing.

Reply:
To be honest, I added this as a placeholder while we settle on where the keyword arguments are supposed to be handled, since we should either import that into TensorAlgebra, or provide a different interface. I just wanted to run some tests in the meantime, but for sure will adapt this :)